<!DOCTYPE html><html lang="zh-CN"><head><meta http-equiv="content-type" content="text/html; charset=utf-8"><meta content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" name="viewport"><meta content="yes" name="apple-mobile-web-app-capable"><meta content="black-translucent" name="apple-mobile-web-app-status-bar-style"><meta content="telephone=no" name="format-detection"><meta name="description" content=""><link rel="stylesheet" type="text/css" href="//fonts.loli.net/css?family=Source+Code+Pro"><link rel="stylesheet" type="text/css" href="/css/style.css?v=2.0.4"><link rel="stylesheet" type="text/css" href="/css/highlight.css?v=2.0.4"><link rel="Shortcut Icon" href="/favicon.ico"><link rel="bookmark" href="/favicon.ico"><link rel="apple-touch-icon" href="/apple-touch-icon.png"><link rel="apple-touch-icon-precomposed" href="/apple-touch-icon.png"><title>利用AutoDOCK进行分子对接 | Billy and Barney's Blog</title><meta name="generator" content="Hexo 4.2.0"></head><body><div class="body_container"><div id="header"><div class="site-name"><h1 class="hidden">利用AutoDOCK进行分子对接</h1><a id="logo" href="/.">Billy and Barney's Blog</a><p class="description"></p></div><div id="nav-menu"><a href="/." class="current"><i class="fa fa-home"> 首页</i></a><a href="/archives/"><i class="fa fa-archive"> 归档</i></a><a href="/about/"><i class="fa fa-user"> 关于</i></a></div><div id="search-form"><div id="result-mask" class="hide"></div><label><input id="search-key" type="text" autocomplete="off" placeholder="搜索"></label><div id="result-wrap" class="hide"><div id="search-result"></div></div><div class="hide"><template id="search-tpl"><div class="item"><a href="/{path}" title="{title}"><div class="title">{title}</div><div class="time">{date}</div><div class="tags">{tags}</div></a></div></template></div></div></div><div id="layout" class="layout-g"><div class="layout-l"><div class="content_container"><div class="post"><h1 class="post-title">利用AutoDOCK进行分子对接</h1><div class="post-meta"><a href="/2019/06/16/Billy--%E5%88%A9%E7%94%A8AutoDOCK%E8%BF%9B%E8%A1%8C%E5%88%86%E5%AD%90%E5%AF%B9%E6%8E%A5/#comments" class="comment-count"></a><p><span class="date">Jun 16, 2019</span><span><a href="/categories/Billy/" class="category">Billy</a></span><span><i id="busuanzi_container_page_pv"><i id="busuanzi_value_page_pv"></i><i>点击</i></i></span></p></div><div class="post-content"><h3 id="分子对接-Moleculer-docking-理论"><a href="#分子对接-Moleculer-docking-理论" class="headerlink" title="分子对接(Moleculer-docking)理论"></a>分子对接(Moleculer-docking)理论</h3><p>分子对接就是两个或多个分子之间通过几何匹配和能量匹配相互识别找到最佳匹配模式的过程。分子对接在酶学研究和药物设计中具有重要的应用意义。</p>
<p>分子对接计算是在受体活性位点区域通过<code>空间结构互补</code>和<code>能量最小化</code>原则来搜寻配体与受体是否能产生相互作用以及它们之间的最佳结合模式。其思想起源于Fisher的锁钥模型(即一把钥匙开一把锁),主要强调的是空间形状的匹配。但配体和受体的识别要比这个模型更加复杂。首先,配体和受体在对接过程中会由于相互适应而产生构象的变化。其次,分子对接还要求能量匹配,对接过程中结合自由能的变化决定了两个分子是否能够结合以及结合的强度。</p>
<p>1958年D.E.Koshland提出分子识别过程中的诱导契合概念,受体分子活性中心的结构原本并非与底物完全吻合,但其是柔软和可塑的。当配体与受体相遇时,可诱导受体构象发生相应的变化,从而便于他们的结合进而引起相应的反应。</p>
<p>分子对接方法根据不同的简化程度分为三类:刚性对接、半柔性对接和柔性对接。刚性对接指在对接过程中,受体和配体的构象不发生变化,适合研究比较大的体系如蛋白-蛋白之间以及蛋白-核酸之间,计算简单,主要考虑对象之间的契合程度。半柔性对接常用于小分子和大分子的对接,在对接过程中,小分子的构象可以在一定范围内变化,但大分子是刚性的。这样既可以在一定程度上考察柔性的影响,又能保持较高的计算效率。在药物设计和虚拟筛选过程中一般采用半柔性的分子对接方法。柔性对接方法一般用于精确研究分子之间的识别情况,由于允许对接体系的构象变化,可以提高对接准确性但耗时较长。</p>
<p>分子对接的目的是找到底物分子和受体分子最佳结合位置及其结合强度,最终可以获得配体和受体的结合构象,但这样的构象可以有很多,一般认为自由能最小的构象存在的概率最高。搜寻最佳构象就要用到构象搜索方法,常用的有系统搜索法和非系统搜索法。系统搜索法通过改变每个扭转角评估所有可能的结合构象,进而选取能量最低的。这一方法计算量非常大。因此通常使用非系统搜索法来寻找能量较低构象,常用方法有:分子动力学方法、随机搜索、遗传算法、距离几何算法等。随机搜索又包括完全随机算法、蒙特卡罗法和模拟退火法等。</p>
<h3 id="分子对接软件包——AutoDock"><a href="#分子对接软件包——AutoDock" class="headerlink" title="分子对接软件包——AutoDock"></a>分子对接软件包——AutoDock</h3><h4 id="AutoDock软件简介"><a href="#AutoDock软件简介" class="headerlink" title="AutoDock软件简介"></a>AutoDock软件简介</h4><p>AutoDOCK是一款由Scripps研究所的Olson实验室开发与维护的开源的分子模拟软件,最主要应用于蛋白与小分子配体的分子对接。其用户图形化界面(GUI)工具为AutoDOCK Tools(ADT)。AutoDock的新一代产品为AutoDock Vina。</p>
<p>AutoDock Vina使用拉马克遗传算法提高效率。软件把遗传算法和局部搜索结合在一起,遗传算法用于全局搜索,而局部搜索用于能量优化。为了加快计算速度,AutoDock Vina采用格点(grid)计算。首先在受体活性氨基酸附近划定一个长方体区域作为搜索空间,扫描不同类型的原子计算格点能量,在搜索空间内,调整配体的构象、位置和方向,进而评分、排序获得能量最低的构象作为输出结果。</p>
<p>对范德华相互作用的计算:每个格点上保存的范德华能量的值的数目与要对接的配体上的原子类型数目相同。如果一个配体中含有C、H、O三种原子类型,那么每个葛店需要用单个探针原子与来计算其与受体之间的范德华相互作用值。当配体与受体进行分子对接时,配体中某个原子和受体之间的相互作用能通过周围8个格点上的这种原子类型为探针的格点值用内插法得到。</p>
<p>静电相互作用的计算采用静电势格点。当配体与受体对接时,某个原子和受体之间的静电相互作用能通过周围格点上静电势以及原子上的部分电荷计算得到。</p>
<h4 id="准备docking需要的受体-蛋白-和配体-化合物"><a href="#准备docking需要的受体-蛋白-和配体-化合物" class="headerlink" title="准备docking需要的受体(蛋白)和配体(化合物)"></a>准备docking需要的受体(蛋白)和配体(化合物)</h4><p>Docking算法需要每个原子带有电荷并且需要标记原子的属性。这些信息通常未包含在PDB文件中。我们需要在对蛋白和小分子的PDB文件预处理,生成同时包含以上信息和PDB文件中原子坐标信息的PDBQT格式文件。进一步地对于“柔性配体docking”,我们还需要定义配体的柔性部分和刚性部分。所有这些都可以通过软件AutoDock Tools (adt)来完成。</p>
<h5 id="准备受体蛋白"><a href="#准备受体蛋白" class="headerlink" title="准备受体蛋白"></a>准备受体蛋白</h5><ol>
<li>PDB文件(1hsg.pdb)中包含了蛋白、配体和水分子;首先提取出蛋白的坐标,即以关键字ATOM和TER开头的行存储到文件1hsg_prot.pdb。<blockquote>
<p>在Linux下,使用命令 egrep “^(ATOM|TER)” 1hsg.pdb &gt; 1hsg_prot.pdb</p>
</blockquote>
</li>
<li>启动AutoDockTools<blockquote>
<p>选择支持GUI的客户端,如MobaXterm,连接服务器。进入AutoDockTools安装目录的’bin’子目录,使用命令 ./adt &amp;</p>
</blockquote>
</li>
<li>依次点选 File–Read Molecule–1hsg_prot.pdb加载蛋白分子。<blockquote>
<p>ADT中按住左键拖动旋转分子结构;点击中键滚动缩放;按住右键移动晶体位置。</p>
</blockquote>
</li>
<li>更改展示方式<blockquote>
<p>依次点选 Color–By Atom Type–All Geometries–OK。</p>
</blockquote>
</li>
<li>加氢:晶体结构中通常缺少氢原子的坐标(因为氢原子电子少,且质子核对电子吸引能力弱,因此很难定位,具体见 <a href="http://www.uh.edu/~chembi/ChemSocRev_Jones_critical.pdf" target="_blank" rel="noopener">http://www.uh.edu/~chembi/ChemSocRev_Jones_critical.pdf</a>  )。但是在docking过程中,氢原子尤其是极性氢原子对计算静电作用是必须的。因此我们需要给蛋白加上氢原子,之后氢原子会以白色短线形式出现。<blockquote>
<p>依次点选 Edit–Hydrogen–Add–Polar only–OK(只加极性H)</p>
</blockquote>
</li>
</ol>
<p><a href="https://i.loli.net/2019/06/16/5d060672b9fbe96584.png" target="_blank" rel="noopener"><img src="https://i.loli.net/2019/06/16/5d060672b9fbe96584.png" alt="1hsg_prot_no_H.png"></a><br><a href="https://i.loli.net/2019/06/16/5d06080c1015a60065.png" target="_blank" rel="noopener"><img src="https://i.loli.net/2019/06/16/5d06080c1015a60065.png" alt="1hsg_prot_with_H.png"></a></p>
<ol start="6">
<li>存储对蛋白的每个原子所做的修改和原子类型判断。<blockquote>
<p>依次点选 Grid–Macromolecule–Choose–1HSG_protein–Slect Molecule (ADT会弹出一个信息框包含程序所做的处理,比如合并非极性氢原子,计算原子局部电荷和判断原子类型,并提示保存Save–1hsg_prot.pdbqt)</p>
</blockquote>
</li>
<li>在受体蛋白定义配体结合的3D搜索空间。如果我们事先不知道结合位点,理论上可以定义一个长方体盒子包含整个蛋白或者随便一个特定区域 (下文PDB文件解析中会提到PDB文件中有时会包含活性位点信息)。<blockquote>
<p>依次点选 Grid – Grid box ,将会在蛋白上画出一个长方体,并且有一个弹出框。在弹出框中,拖拽刻度线查看长方体的变化,完成设置。<br>在这个例子中,我们知道结合位点,就选取以其为中心的一个小空间。<br>设置Spacing (angstrom)为1埃 (这实际是一个换算系数, 相当于步长; 默认为0.375,是C-C单键长度的1/4,最大为1。spacing值与(各个维度上的点的数目+1)的乘积就是长方体Grid box的大小)。在我们调整的过程中,可以看到随着这个数值的变大,立方体也被放大了。<br>另外我们设置x,y,z center为16,25,4,number of points in (x,y,z)-dimension为30,30,30(最大为126,必须为偶数,AutoDock会自动再每一维再加一个点)。记下我们设置的这些点,下面会用到。在刻度转盘处点击右键会弹出一个窗口,输入数字回车即可设置GRID的中心坐标和大小。较大的number of points in (xyz)-dimension和较小的Spacing会增加搜索的精度,同时需要花费更多的计算时间。</p>
</blockquote>
</li>
<li>设置受体的柔性残基:<blockquote>
<p>在ADT中依次点选 Flexible Residues–Input–Choose Macromolecule–1hsg_prot; select–select from string–Residue: ARG8–Add–Dismiss, 8号ARG氨基酸残基就被选中了。<br>再依次点选Flexible Residues–Choose Torsions in Currently Selected Residues将选择的残基标记为柔性残基并设置可扭转的数量。<br>在分子显示窗口中分别点击两个残基上CA和CB原子之间的建,使之变为非扭转的(紫色显示)[Shift + 鼠标左键],这样两个残基中的32个键中有6个是可扭转的。这里设置配体的柔性残基或者使CA-CB的键为刚性都是可选操作。<br>放在教程中只是用来展示怎么操作的,无其它指导意义。Flexible Residues–Output–Save Flexible PDBQT保存柔性残基文件。 Flexible Residues–Output–Save Rigid PDBQT保存柔性残基文件。</p>
</blockquote>
</li>
<li>关掉grid和删除protein<blockquote>
<p>Grid Options–File–Close w/out saving; Edit–Delete–Delete Molecule–1hsg_prot–Continue</p>
</blockquote>
</li>
</ol>
<h5 id="准备配体"><a href="#准备配体" class="headerlink" title="准备配体"></a>准备配体</h5><p>与蛋白结构类似,配体的结构也缺少氢原子,我们需要添加氢原子并且定义哪些键是可以旋转的以用于柔性docking。</p>
<ol>
<li>从PDB结构中提取配体的原子位置。indinavir的配体残基名字为MK1,以HETATM开头的行表示非核心多聚体的成分 (heteroatoms)(具体见PDB文件格式解释)。<blockquote>
<p>Linux系统下,运行 grep “^HETATM.*MK1” 1hsg.pdb &gt; indinavir.pdb</p>
</blockquote>
</li>
<li>将结构读入ADT:<blockquote>
<p>依次点选 File–Read Molecule–indinavir.pdb<br>依次点选 Color–By Atom Type–All Geometreies–OK</p>
</blockquote>
</li>
<li>给配体加上氢原子:<blockquote>
<p>Edit–Hydrogen–Add–Polar only–OK</p>
</blockquote>
</li>
<li>在ADT中定义此化合物为配体,以便ADT为其计算局部电荷(partial charges)和设置可旋转配体键。<blockquote>
<p>依次点选 Ligand–input–Choose–indinavir–Select Molecule for AutoDock4<br>这时会有一个弹出框显示ADT所做的操作,包括合并非极性氢(只在添加了的情况下)、计算电荷电量和设置旋转键。<br>然后点选 Ligand–Output–Save as PDBQT 存储结果。</p>
</blockquote>
</li>
<li>查看ADT检测出的旋转键:<blockquote>
<p>依次点选 Ligand–Torsion Tree–Choose Torsions,可以看到Number of rotatable bonds=14/32。</p>
</blockquote>
</li>
</ol>
<h5 id="准备docking配置文件"><a href="#准备docking配置文件" class="headerlink" title="准备docking配置文件"></a>准备docking配置文件</h5><p>docking配置文件包含了输入的受体(蛋白)、配体(化合物)和搜索参数的信息,为一个文本文件,名字任意,可以为conf.txt,内容如下:</p>
<blockquote>
<p>receptor = 1hsg_prot.pdbqt<br>ligand   = indinavir.pdbqt<br>num_modes = 50<br>out = dockingResult.pdbqt<br>log = docking.log<br>center_x = 16<br>center_y = 25<br>center_z = 4<br>size_x = 30<br>size_y = 30<br>size_z = 30<br>seed = 2009</p>
</blockquote>
<blockquote>
<p>receptor和ligand为输入文件的名字,与conf.txt在同一目录下; out为输出文件的名字;log为输出日志文件的名字。<br>centerhe和size定义搜索空间的位置和大小。<br>num_modes设置最多显示的结合模型,鉴于只输出符合能量值要求的结果,最后输出的结合模型数量可能少于这一数值。<br>seed设置随机数生成的种子,可以为任意整数。如果想重现之前的分析结果就需要使用相同的seed。</p>
</blockquote>
<h5 id="Docking-小分子化合物indinavir到HIV-1蛋白酶"><a href="#Docking-小分子化合物indinavir到HIV-1蛋白酶" class="headerlink" title="Docking 小分子化合物indinavir到HIV-1蛋白酶"></a>Docking 小分子化合物indinavir到HIV-1蛋白酶</h5><ol>
<li>使用AutoDock Vina执行docking预测<blockquote>
<p>在Linux终端执行 path_to_vina/vina –config conf.txt</p>
</blockquote>
</li>
<li>输出结果包含两个文件,构象文件dockingResult.pdbqt和日志文件docking.log.<blockquote>
<p>dockingResult.pdbqt: 包含所有docking的模式,通常第一个为结合最好的构象,但如果前几个能量值相差不大时也有例外。<br>docking.log: 日志文件,包含结合能量值(第一列,越低越稳定,默认由低到高排序,所以第一个为最好的构象)、每个构象与第一个构象的距离、每个构象与第一个构象的差别。</p>
</blockquote>
</li>
</ol>
<h5 id="用Pymol可视化结果"><a href="#用Pymol可视化结果" class="headerlink" title="用Pymol可视化结果"></a>用Pymol可视化结果</h5><p>打开PyMOL,依次点选File–Open文件类型选择All Files-选取结果dockingResult.pdbqt文件、原始蛋白和配体的pdb文件、原教程的pdbqt文件。</p>
<h5 id="用Pymol展示配体和受体相互作用的原子和氢键"><a href="#用Pymol展示配体和受体相互作用的原子和氢键" class="headerlink" title="用Pymol展示配体和受体相互作用的原子和氢键"></a>用Pymol展示配体和受体相互作用的原子和氢键</h5><p>为了简化展示过程,我们设计了一个pml脚本 (脚本内有很详细的解释),只需要修改脚本里面受体和配体的名字,然后在PyMOL的命令行界面输入PyMOL&gt; run display.pml即可获得展示结果。当然这个脚本也可以使用程序generatePmlForHbond.py生成。</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br><span class="line">146</span><br><span class="line">147</span><br><span class="line">148</span><br><span class="line">149</span><br><span class="line">150</span><br><span class="line">151</span><br><span class="line">152</span><br><span class="line">153</span><br><span class="line">154</span><br><span class="line">155</span><br><span class="line">156</span><br><span class="line">157</span><br><span class="line">158</span><br></pre></td><td class="code"><pre><span class="line"> ############################################################</span><br><span class="line">###All one needs to do is replacing:                        ##</span><br><span class="line">###  * Protein structure file: E:\docking\1hsg_prot.pdb     ##</span><br><span class="line">###  * Protein name: 1hsg                                   ##</span><br><span class="line">###  * Docking result file: E:\docking\indinavir.pdbqt      ##</span><br><span class="line">###  * Docking result name (normally ligand name): indinavir##</span><br><span class="line"> ############################################################</span><br><span class="line"># The following 4 lines: </span><br><span class="line">	# 1. load protein structure and rename it</span><br><span class="line">	# 2. add hydrogen (&#96;h_add&#96; uses a primitive algorithm to add hydrogens onto a molecule.)</span><br><span class="line">	# 3. hide protein display</span><br><span class="line">	# 4. show cartoon display for protein</span><br><span class="line">load E:\yunpan\docking\1hsg_prot.pdb, 1hsg</span><br><span class="line">h_add 1hsg</span><br><span class="line">hide everything, 1hsg</span><br><span class="line">show cartoon, 1hsg</span><br><span class="line">cmd.spectrum(&quot;count&quot;, selection&#x3D;&quot;1hsg&quot;, byres&#x3D;1)</span><br><span class="line"></span><br><span class="line"># The following 6 lines: </span><br><span class="line">	# 1. load ligand structure and rename it</span><br><span class="line">	# 2. add hydrogen</span><br><span class="line">	# 3. hide ligand display</span><br><span class="line">	# 4. show ligand in sticks mode</span><br><span class="line">	# 5. Set width of stick to 0.15</span><br><span class="line">	# 6. Set atom color: C-white;N-blue;O-red</span><br><span class="line">load E:\yunpan\docking\indinavir.pdbqt, indinavir</span><br><span class="line">h_add indinavir</span><br><span class="line">hide everything, indinavir</span><br><span class="line">show sticks, indinavir</span><br><span class="line">set stick_radius, 0.15</span><br><span class="line">util.cbaw indinavir</span><br><span class="line"></span><br><span class="line"># The following 1 line:</span><br><span class="line">	# 1. Select metal ions</span><br><span class="line">select metals, symbol mg+ca+fe+zn</span><br><span class="line"></span><br><span class="line"># The following 2 lines:</span><br><span class="line">	# 1. Set hydrogen donator</span><br><span class="line">	# 2. Set hydrogen accrptor </span><br><span class="line">	# &#96;select&#96; creates a named selection from an atom selection. </span><br><span class="line">	# &#96;select name, (selection)&#96;</span><br><span class="line">select h_donator,  (elem n,o and (neighbor hydro))</span><br><span class="line">select h_acceptor, (elem o or (elem n and not (neighbor hydro)))</span><br><span class="line"></span><br><span class="line"># The following 4 lines:</span><br><span class="line">	# 1. Create link between ligand_h_acceptor and prot_h_donator  within given distance 3.2</span><br><span class="line">	# 2. Create link between ligand_h_donator  and prot_h_acceptor within given distance 3.2</span><br><span class="line">	#    Set filter 3.6 for ideal geometry and filter 3.2 for minimally acceptable geometry</span><br><span class="line">	# 3. Set red color for ligand_h_acceptor and prot_h_donator </span><br><span class="line">	# 4. Set blue color for ligand_h_donator  and prot_h_acceptor</span><br><span class="line">	# &#96;distance&#96; creates a new distance object between two selections. It will display all distances within the cutoff. Distance is also used to make hydrogen bonds like &#96;distance hbonds, all, all, 3.2, mode&#x3D;2&#96;.</span><br><span class="line">	# distance [ name [, selection1 [, selection2 [, cutoff [, mode ]]]]]</span><br><span class="line">distance LaccPdon, (indinavir and h_acceptor), (1hsg and h_donator), 3.2</span><br><span class="line">distance LdonPacc, (indinavir and h_donator), (1hsg and h_acceptor), 3.2</span><br><span class="line">color red, LaccPdon</span><br><span class="line">color blue, LdonPacc</span><br><span class="line">#distance Fe_C20, (fep and name C20), (heme and name fe))</span><br><span class="line"></span><br><span class="line"># The following 6 lines:</span><br><span class="line">	# 1. Select non-hydro atoms of ligands</span><br><span class="line">	# 2. Select protein atoms within 5A of selected atoms in last step</span><br><span class="line">	# 3. Label alpha-c(ca) of selected residues with residue name and residue position</span><br><span class="line">	# 4. Set label color back</span><br><span class="line">	# 5. Set background white</span><br><span class="line">	# 6. Hidden hydrogenes</span><br><span class="line">select sele, indinavir &amp; not hydro</span><br><span class="line">select sele, byres (sele expand 5) &amp; 1hsg</span><br><span class="line">one_letter &#x3D;&#123;&#39;VAL&#39;:&#39;V&#39;, &#39;ILE&#39;:&#39;I&#39;, &#39;LEU&#39;:&#39;L&#39;, &#39;GLU&#39;:&#39;E&#39;, &#39;GLN&#39;:&#39;Q&#39;, \</span><br><span class="line">&#39;ASP&#39;:&#39;D&#39;, &#39;ASN&#39;:&#39;N&#39;, &#39;HIS&#39;:&#39;H&#39;, &#39;TRP&#39;:&#39;W&#39;, &#39;PHE&#39;:&#39;F&#39;, &#39;TYR&#39;:&#39;Y&#39;,    \</span><br><span class="line">&#39;ARG&#39;:&#39;R&#39;, &#39;LYS&#39;:&#39;K&#39;, &#39;SER&#39;:&#39;S&#39;, &#39;THR&#39;:&#39;T&#39;, &#39;MET&#39;:&#39;M&#39;, &#39;ALA&#39;:&#39;A&#39;,    \</span><br><span class="line">&#39;GLY&#39;:&#39;G&#39;, &#39;PRO&#39;:&#39;P&#39;, &#39;CYS&#39;:&#39;C&#39;&#125;</span><br><span class="line">label name ca &amp; sele, &quot;%s-%s&quot; % (one_letter[resn],resi)</span><br><span class="line">bg white</span><br><span class="line">set label_color, black</span><br><span class="line">hide (hydro)</span><br><span class="line"></span><br><span class="line"># The follwing 5 lines</span><br><span class="line">	# 1. Comment out this line</span><br><span class="line">	# 2. Create an object &#96;surrounding_res&#96; to represent selected protein atoms</span><br><span class="line">	#    &#96;create&#96;: creates a new molecule object from a selection. It can also be used to create states in an   existing object. </span><br><span class="line">	#    &#96;create name, (selection)&#96;</span><br><span class="line">	# 3. Display created surface</span><br><span class="line">	# 4. Set color for surrounding_res</span><br><span class="line">	# 5. Set transparency for surrounding_res</span><br><span class="line">	#    Transparency is used to adjust the transparency of Surfaces and Slices.    </span><br><span class="line">	#    &#96;set transparency, F, selection&#96;</span><br><span class="line">#show surface, 1hsg</span><br><span class="line">create surrounding_res, sele</span><br><span class="line">show surface, surrounding_res</span><br><span class="line">color grey80, surrounding_res</span><br><span class="line">set transparency, 0.5, surrounding_res</span><br><span class="line"></span><br><span class="line">此外还可以使用如下脚本(list_hbonds.py)输出相互作用的原子及其位置。</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"># Copyright (c) 2010 Robert L. Campbell</span><br><span class="line">from pymol import cmd</span><br><span class="line"></span><br><span class="line">def list_hb(selection,selection2&#x3D;None,cutoff&#x3D;3.2,angle&#x3D;55,mode&#x3D;1,hb_list_name&#x3D;&#39;hbonds&#39;):</span><br><span class="line">    &quot;&quot;&quot;</span><br><span class="line">    USAGE</span><br><span class="line">  </span><br><span class="line">    list_hb selection, [selection2 (default&#x3D;None)], [cutoff (default&#x3D;3.2)],</span><br><span class="line">                       [angle (default&#x3D;55)], [mode (default&#x3D;1)],</span><br><span class="line">                       [hb_list_name (default&#x3D;&#39;hbonds&#39;)]</span><br><span class="line">  </span><br><span class="line">    The script automatically adds a requirement that atoms in the</span><br><span class="line">    selection (and selection2 if used) must be either of the elements N or</span><br><span class="line">    O.</span><br><span class="line">  </span><br><span class="line">    If mode is set to 0 instead of the default value 1, then no angle</span><br><span class="line">    cutoff is used, otherwise the angle cutoff is used and defaults to 55</span><br><span class="line">    degrees.</span><br><span class="line">  </span><br><span class="line">    e.g.</span><br><span class="line">    To get a list of all H-bonds within chain A of an object</span><br><span class="line">      list_hb 1abc &amp; c. a &amp;! r. hoh, cutoff&#x3D;3.2, hb_list_name&#x3D;abc-hbonds</span><br><span class="line">  </span><br><span class="line">    To get a list of H-bonds between chain B and everything else:</span><br><span class="line">      list_hb 1tl9 &amp; c. b, 1tl9 &amp;! c. b</span><br><span class="line">  </span><br><span class="line">    &quot;&quot;&quot;</span><br><span class="line">    cutoff&#x3D;float(cutoff)</span><br><span class="line">    angle&#x3D;float(angle)</span><br><span class="line">    mode&#x3D;float(mode)</span><br><span class="line">    # ensure only N and O atoms are in the selection</span><br><span class="line">    selection &#x3D; selection + &quot; &amp; e. n+o&quot;</span><br><span class="line">    if not selection2:</span><br><span class="line">        hb &#x3D; cmd.find_pairs(selection,selection,mode&#x3D;mode,cutoff&#x3D;cutoff,angle&#x3D;angle)</span><br><span class="line">    else:</span><br><span class="line">        selection2 &#x3D; selection2 + &quot; &amp; e. n+o&quot;</span><br><span class="line">        hb &#x3D; cmd.find_pairs(selection,selection2,mode&#x3D;mode,cutoff&#x3D;cutoff,angle&#x3D;angle)</span><br><span class="line">  </span><br><span class="line">    # sort the list for easier reading</span><br><span class="line">    hb.sort(lambda x,y:(cmp(x[0][1],y[0][1])))</span><br><span class="line">  </span><br><span class="line">    for pairs in hb:</span><br><span class="line">        cmd.iterate(&quot;%s and index %s&quot; % (pairs[0][0],pairs[0][1]), &#39;print &quot;%1s&#x2F;%3s&#96;%s&#x2F;%-4s &quot; % (chain,resn,resi,name),&#39;)</span><br><span class="line">        cmd.iterate(&quot;%s and index %s&quot; % (pairs[1][0],pairs[1][1]), &#39;print &quot;%1s&#x2F;%3s&#96;%s&#x2F;%-4s &quot; % (chain,resn,resi,name),&#39;)</span><br><span class="line">        print &quot;%.2f&quot; % cmd.distance(hb_list_name,&quot;%s and index %s&quot; % (pairs[0][0],pairs[0][1]),&quot;%s and index %s&quot; % (pairs[1][0],pairs[1][1]))</span><br><span class="line"></span><br><span class="line">#cmd.extend(&quot;list_hb&quot;,list_hb)</span><br><span class="line">#if __name__ &#x3D;&#x3D; &quot;__main__&quot;:</span><br><span class="line">cmd.load(&quot;E:&#x2F;yunpan&#x2F;docking&#x2F;1hsg_prot.pdb&quot;, &quot;1hsg&quot;)</span><br><span class="line">cmd.h_add(&quot;(1hsg)&quot;)</span><br><span class="line">cmd.load(&quot;E:&#x2F;yunpan&#x2F;docking&#x2F;indinavir.pdbqt&quot;,&quot;indinavir&quot;)</span><br><span class="line">cmd.h_add(&quot;(indinavir)&quot;)</span><br><span class="line"></span><br><span class="line">h_donator  &#x3D; &quot;elem n,o &amp; (neighbor hydro)&quot;</span><br><span class="line">h_acceptor &#x3D; &quot;elem o | (elem n &amp; !(neighbor hydro))&quot;</span><br><span class="line"></span><br><span class="line">lacc &#x3D; &quot;indinavir &amp; (elem o | (elem n &amp; !(neighbor hydro)))&quot;</span><br><span class="line">ldon &#x3D; &quot;indinavir &amp; (elem n,o &amp; (neighbor hydro))&quot;</span><br><span class="line">pacc &#x3D; &quot;1hsg &amp; (elem o | (elem n &amp; !(neighbor hydro)))&quot;</span><br><span class="line">pdon &#x3D; &quot;1hsg &amp; (elem n,o &amp; (neighbor hydro))&quot;</span><br><span class="line"></span><br><span class="line">list_hb(ldon, pacc, hb_list_name&#x3D;&quot;l2p_hbonds&quot;)</span><br><span class="line">list_hb(lacc, pdon, hb_list_name&#x3D;&quot;p2l_hbonds&quot;)</span><br></pre></td></tr></table></figure>
<blockquote>
<p>输出结果如下:<br>PyMOL&gt;run E:/docking/list_hbonds.py<br>B/MK1’902/N4    B/GLY’27/O     3.03<br>B/MK1’902/O4    B/GLY’27/O     3.16<br>B/MK1’902/O2    A/ASP’25/OD1   2.77<br>B/MK1’902/O2    B/ASP’25/OD1   2.63</p>
</blockquote>
<p>看上去比显示的氢键少了三个,这是因为我们在第二个函数中使用了H-键角度限制,如果在调用时给定参数list_hb(mode=0)则会获得一致结果。<br><a href="https://i.loli.net/2019/06/16/5d0620a96b27533399.png" target="_blank" rel="noopener"><img src="https://i.loli.net/2019/06/16/5d0620a96b27533399.png" alt="after_run_list_hbond.png"></a><br><a href="https://i.loli.net/2019/06/16/5d0620a9e191835511.png" target="_blank" rel="noopener"><img src="https://i.loli.net/2019/06/16/5d0620a9e191835511.png" alt="after_run_hbond.pml.png"></a><br>H-bond结果展示。第一张图为运行<code>display.pml</code>后的结果,蓝色虚线为氢键;第二张图为运行<code>list_hbonds.py</code>后的结果, 黄色虚线为氢键(覆盖了之前的蓝色)。可以通过点选<code>LaccPdon</code>, <code>LdonPacc</code>, <code>l2p_hbonds</code>显示不同的氢键。</p>
<h5 id="展示疏水表面"><a href="#展示疏水表面" class="headerlink" title="展示疏水表面"></a>展示疏水表面</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br></pre></td><td class="code"><pre><span class="line"># color_h</span><br><span class="line"># -------</span><br><span class="line"> </span><br><span class="line"># PyMOL command to color protein molecules according to the Eisenberg hydrophobicity scale</span><br><span class="line"> </span><br><span class="line">#</span><br><span class="line"># Source: http:&#x2F;&#x2F;us.expasy.org&#x2F;tools&#x2F;pscale&#x2F;Hphob.Eisenberg.html</span><br><span class="line"># Amino acid scale: Normalized consensus hydrophobicity scale</span><br><span class="line"># Author(s): Eisenberg D., Schwarz E., Komarony M., Wall R.</span><br><span class="line"># Reference: J. Mol. Biol. 179:125-142 (1984)</span><br><span class="line">#</span><br><span class="line"># Amino acid scale values:</span><br><span class="line">#</span><br><span class="line"># Ala:  0.620</span><br><span class="line"># Arg: -2.530</span><br><span class="line"># Asn: -0.780</span><br><span class="line"># Asp: -0.900</span><br><span class="line"># Cys:  0.290</span><br><span class="line"># Gln: -0.850</span><br><span class="line"># Glu: -0.740</span><br><span class="line"># Gly:  0.480</span><br><span class="line"># His: -0.400</span><br><span class="line"># Ile:  1.380</span><br><span class="line"># Leu:  1.060</span><br><span class="line"># Lys: -1.500</span><br><span class="line"># Met:  0.640</span><br><span class="line"># Phe:  1.190</span><br><span class="line"># Pro:  0.120</span><br><span class="line"># Ser: -0.180</span><br><span class="line"># Thr: -0.050</span><br><span class="line"># Trp:  0.810</span><br><span class="line"># Tyr:  0.260</span><br><span class="line"># Val:  1.080</span><br><span class="line">#</span><br><span class="line"># Usage:</span><br><span class="line"># color_h (selection)</span><br><span class="line">#</span><br><span class="line">from pymol import cmd</span><br><span class="line"> </span><br><span class="line">def color_h(selection&#x3D;&#39;all&#39;):</span><br><span class="line">        s &#x3D; str(selection)</span><br><span class="line">	print s</span><br><span class="line">        cmd.set_color(&#39;color_ile&#39;,[0.996,0.062,0.062])</span><br><span class="line">        cmd.set_color(&#39;color_phe&#39;,[0.996,0.109,0.109])</span><br><span class="line">        cmd.set_color(&#39;color_val&#39;,[0.992,0.156,0.156])</span><br><span class="line">        cmd.set_color(&#39;color_leu&#39;,[0.992,0.207,0.207])</span><br><span class="line">        cmd.set_color(&#39;color_trp&#39;,[0.992,0.254,0.254])</span><br><span class="line">        cmd.set_color(&#39;color_met&#39;,[0.988,0.301,0.301])</span><br><span class="line">        cmd.set_color(&#39;color_ala&#39;,[0.988,0.348,0.348])</span><br><span class="line">        cmd.set_color(&#39;color_gly&#39;,[0.984,0.394,0.394])</span><br><span class="line">        cmd.set_color(&#39;color_cys&#39;,[0.984,0.445,0.445])</span><br><span class="line">        cmd.set_color(&#39;color_tyr&#39;,[0.984,0.492,0.492])</span><br><span class="line">        cmd.set_color(&#39;color_pro&#39;,[0.980,0.539,0.539])</span><br><span class="line">        cmd.set_color(&#39;color_thr&#39;,[0.980,0.586,0.586])</span><br><span class="line">        cmd.set_color(&#39;color_ser&#39;,[0.980,0.637,0.637])</span><br><span class="line">        cmd.set_color(&#39;color_his&#39;,[0.977,0.684,0.684])</span><br><span class="line">        cmd.set_color(&#39;color_glu&#39;,[0.977,0.730,0.730])</span><br><span class="line">        cmd.set_color(&#39;color_asn&#39;,[0.973,0.777,0.777])</span><br><span class="line">        cmd.set_color(&#39;color_gln&#39;,[0.973,0.824,0.824])</span><br><span class="line">        cmd.set_color(&#39;color_asp&#39;,[0.973,0.875,0.875])</span><br><span class="line">        cmd.set_color(&#39;color_lys&#39;,[0.899,0.922,0.922])</span><br><span class="line">        cmd.set_color(&#39;color_arg&#39;,[0.899,0.969,0.969])</span><br><span class="line">        cmd.color(&quot;color_ile&quot;,&quot;(&quot;+s+&quot; and resn ile)&quot;)</span><br><span class="line">        cmd.color(&quot;color_phe&quot;,&quot;(&quot;+s+&quot; and resn phe)&quot;)</span><br><span class="line">        cmd.color(&quot;color_val&quot;,&quot;(&quot;+s+&quot; and resn val)&quot;)</span><br><span class="line">        cmd.color(&quot;color_leu&quot;,&quot;(&quot;+s+&quot; and resn leu)&quot;)</span><br><span class="line">        cmd.color(&quot;color_trp&quot;,&quot;(&quot;+s+&quot; and resn trp)&quot;)</span><br><span class="line">        cmd.color(&quot;color_met&quot;,&quot;(&quot;+s+&quot; and resn met)&quot;)</span><br><span class="line">        cmd.color(&quot;color_ala&quot;,&quot;(&quot;+s+&quot; and resn ala)&quot;)</span><br><span class="line">        cmd.color(&quot;color_gly&quot;,&quot;(&quot;+s+&quot; and resn gly)&quot;)</span><br><span class="line">        cmd.color(&quot;color_cys&quot;,&quot;(&quot;+s+&quot; and resn cys)&quot;)</span><br><span class="line">        cmd.color(&quot;color_tyr&quot;,&quot;(&quot;+s+&quot; and resn tyr)&quot;)</span><br><span class="line">        cmd.color(&quot;color_pro&quot;,&quot;(&quot;+s+&quot; and resn pro)&quot;)</span><br><span class="line">        cmd.color(&quot;color_thr&quot;,&quot;(&quot;+s+&quot; and resn thr)&quot;)</span><br><span class="line">        cmd.color(&quot;color_ser&quot;,&quot;(&quot;+s+&quot; and resn ser)&quot;)</span><br><span class="line">        cmd.color(&quot;color_his&quot;,&quot;(&quot;+s+&quot; and resn his)&quot;)</span><br><span class="line">        cmd.color(&quot;color_glu&quot;,&quot;(&quot;+s+&quot; and resn glu)&quot;)</span><br><span class="line">        cmd.color(&quot;color_asn&quot;,&quot;(&quot;+s+&quot; and resn asn)&quot;)</span><br><span class="line">        cmd.color(&quot;color_gln&quot;,&quot;(&quot;+s+&quot; and resn gln)&quot;)</span><br><span class="line">        cmd.color(&quot;color_asp&quot;,&quot;(&quot;+s+&quot; and resn asp)&quot;)</span><br><span class="line">        cmd.color(&quot;color_lys&quot;,&quot;(&quot;+s+&quot; and resn lys)&quot;)</span><br><span class="line">        cmd.color(&quot;color_arg&quot;,&quot;(&quot;+s+&quot; and resn arg)&quot;)</span><br><span class="line">cmd.extend(&#39;color_h&#39;,color_h)</span><br><span class="line"> </span><br><span class="line">def color_h2(selection&#x3D;&#39;all&#39;):</span><br><span class="line">        s &#x3D; str(selection)</span><br><span class="line">	print s</span><br><span class="line">	cmd.set_color(&quot;color_ile2&quot;,[0.938,1,0.938])</span><br><span class="line">	cmd.set_color(&quot;color_phe2&quot;,[0.891,1,0.891])</span><br><span class="line">	cmd.set_color(&quot;color_val2&quot;,[0.844,1,0.844])</span><br><span class="line">	cmd.set_color(&quot;color_leu2&quot;,[0.793,1,0.793])</span><br><span class="line">	cmd.set_color(&quot;color_trp2&quot;,[0.746,1,0.746])</span><br><span class="line">	cmd.set_color(&quot;color_met2&quot;,[0.699,1,0.699])</span><br><span class="line">	cmd.set_color(&quot;color_ala2&quot;,[0.652,1,0.652])</span><br><span class="line">	cmd.set_color(&quot;color_gly2&quot;,[0.606,1,0.606])</span><br><span class="line">	cmd.set_color(&quot;color_cys2&quot;,[0.555,1,0.555])</span><br><span class="line">	cmd.set_color(&quot;color_tyr2&quot;,[0.508,1,0.508])</span><br><span class="line">	cmd.set_color(&quot;color_pro2&quot;,[0.461,1,0.461])</span><br><span class="line">	cmd.set_color(&quot;color_thr2&quot;,[0.414,1,0.414])</span><br><span class="line">	cmd.set_color(&quot;color_ser2&quot;,[0.363,1,0.363])</span><br><span class="line">	cmd.set_color(&quot;color_his2&quot;,[0.316,1,0.316])</span><br><span class="line">	cmd.set_color(&quot;color_glu2&quot;,[0.27,1,0.27])</span><br><span class="line">	cmd.set_color(&quot;color_asn2&quot;,[0.223,1,0.223])</span><br><span class="line">	cmd.set_color(&quot;color_gln2&quot;,[0.176,1,0.176])</span><br><span class="line">	cmd.set_color(&quot;color_asp2&quot;,[0.125,1,0.125])</span><br><span class="line">	cmd.set_color(&quot;color_lys2&quot;,[0.078,1,0.078])</span><br><span class="line">	cmd.set_color(&quot;color_arg2&quot;,[0.031,1,0.031])</span><br><span class="line">        cmd.color(&quot;color_ile2&quot;,&quot;(&quot;+s+&quot; and resn ile)&quot;)</span><br><span class="line">        cmd.color(&quot;color_phe2&quot;,&quot;(&quot;+s+&quot; and resn phe)&quot;)</span><br><span class="line">        cmd.color(&quot;color_val2&quot;,&quot;(&quot;+s+&quot; and resn val)&quot;)</span><br><span class="line">        cmd.color(&quot;color_leu2&quot;,&quot;(&quot;+s+&quot; and resn leu)&quot;)</span><br><span class="line">        cmd.color(&quot;color_trp2&quot;,&quot;(&quot;+s+&quot; and resn trp)&quot;)</span><br><span class="line">        cmd.color(&quot;color_met2&quot;,&quot;(&quot;+s+&quot; and resn met)&quot;)</span><br><span class="line">        cmd.color(&quot;color_ala2&quot;,&quot;(&quot;+s+&quot; and resn ala)&quot;)</span><br><span class="line">        cmd.color(&quot;color_gly2&quot;,&quot;(&quot;+s+&quot; and resn gly)&quot;)</span><br><span class="line">        cmd.color(&quot;color_cys2&quot;,&quot;(&quot;+s+&quot; and resn cys)&quot;)</span><br><span class="line">        cmd.color(&quot;color_tyr2&quot;,&quot;(&quot;+s+&quot; and resn tyr)&quot;)</span><br><span class="line">        cmd.color(&quot;color_pro2&quot;,&quot;(&quot;+s+&quot; and resn pro)&quot;)</span><br><span class="line">        cmd.color(&quot;color_thr2&quot;,&quot;(&quot;+s+&quot; and resn thr)&quot;)</span><br><span class="line">        cmd.color(&quot;color_ser2&quot;,&quot;(&quot;+s+&quot; and resn ser)&quot;)</span><br><span class="line">        cmd.color(&quot;color_his2&quot;,&quot;(&quot;+s+&quot; and resn his)&quot;)</span><br><span class="line">        cmd.color(&quot;color_glu2&quot;,&quot;(&quot;+s+&quot; and resn glu)&quot;)</span><br><span class="line">        cmd.color(&quot;color_asn2&quot;,&quot;(&quot;+s+&quot; and resn asn)&quot;)</span><br><span class="line">        cmd.color(&quot;color_gln2&quot;,&quot;(&quot;+s+&quot; and resn gln)&quot;)</span><br><span class="line">        cmd.color(&quot;color_asp2&quot;,&quot;(&quot;+s+&quot; and resn asp)&quot;)</span><br><span class="line">        cmd.color(&quot;color_lys2&quot;,&quot;(&quot;+s+&quot; and resn lys)&quot;)</span><br><span class="line">        cmd.color(&quot;color_arg2&quot;,&quot;(&quot;+s+&quot; and resn arg)&quot;)</span><br><span class="line">cmd.extend(&#39;color_h2&#39;,color_h2)</span><br></pre></td></tr></table></figure>
<p>将上面的脚本存储为color_h.py,在PyMOL界面运行File–Run–color_h.py,在命令行输入</p>
<blockquote>
<p>PyMOl color_h –Show surface</p>
</blockquote>
</div><div class="post-copyright"><blockquote><p>原文作者: Billy &amp; Barney</p><p>原文链接: <a href="http://yoursite.com/2019/06/16/Billy--利用AutoDOCK进行分子对接/">http://yoursite.com/2019/06/16/Billy--利用AutoDOCK进行分子对接/</a></p><p>版权声明: 转载请注明出处(必须保留作者署名及链接)</p></blockquote></div><div class="tags"><a href="/tags/%E8%AE%A1%E7%AE%97%E6%9C%BA%E8%BE%85%E5%8A%A9%E8%8D%AF%E7%89%A9%E8%AE%BE%E8%AE%A1/">计算机辅助药物设计</a><a href="/tags/%E8%BD%AF%E4%BB%B6%E6%95%99%E7%A8%8B/">软件教程</a></div><div class="post-share"><div class="social-share"><span>分享到:</span></div></div><div class="post-nav"><a href="/2019/06/16/Billy--%E8%BD%AF%E4%BB%B6samtools%E6%95%99%E7%A8%8B%EF%BC%88%E8%BD%AC%E8%BD%BD%EF%BC%89/" class="pre">samtools常用命令详解(转载)</a><a href="/2019/06/02/Billy--Markdown%E5%AD%A6%E4%B9%A0%E7%AC%94%E8%AE%B0/" class="next">Markdown 学习笔记</a></div><div id="comments"></div></div></div></div><div class="layout-r"><div id="sidebar"><div class="search-pla"></div><div id="toc" class="widget"><div class="widget-title"><i class="fa fa-fei">文章目录</i></div><ol class="toc"><li class="toc-item toc-level-3"><a class="toc-link" href="#分子对接-Moleculer-docking-理论"><span class="toc-text">分子对接(Moleculer-docking)理论</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#分子对接软件包——AutoDock"><span class="toc-text">分子对接软件包——AutoDock</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#AutoDock软件简介"><span class="toc-text">AutoDock软件简介</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#准备docking需要的受体-蛋白-和配体-化合物"><span class="toc-text">准备docking需要的受体(蛋白)和配体(化合物)</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#准备受体蛋白"><span class="toc-text">准备受体蛋白</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#准备配体"><span class="toc-text">准备配体</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#准备docking配置文件"><span class="toc-text">准备docking配置文件</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Docking-小分子化合物indinavir到HIV-1蛋白酶"><span class="toc-text">Docking 小分子化合物indinavir到HIV-1蛋白酶</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#用Pymol可视化结果"><span class="toc-text">用Pymol可视化结果</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#用Pymol展示配体和受体相互作用的原子和氢键"><span class="toc-text">用Pymol展示配体和受体相互作用的原子和氢键</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#展示疏水表面"><span class="toc-text">展示疏水表面</span></a></li></ol></li></ol></li></ol></div><div class="widget"><div class="widget-title"><i class="fa fa-xie"> 最新文章</i></div><ul class="post-list"><li class="post-list-item"><a class="post-list-link" href="/2020/01/10/Billy--R%E8%AF%AD%E8%A8%80%E7%9A%84Jupyter%20notebook%E7%8E%AF%E5%A2%83%E6%90%AD%E5%BB%BA/">R语音的Jupyter notebook环境搭建</a></li><li class="post-list-item"><a class="post-list-link" href="/2020/01/05/Barnery--JS%E4%B8%AD%E7%9A%84%E4%BA%8B%E4%BB%B6%E6%9C%BA%E5%88%B6/">JS中的事件机制</a></li><li class="post-list-item"><a class="post-list-link" href="/2020/01/05/BIlly--%E8%82%BF%E7%98%A4%E6%96%B0%E7%94%9F%E6%8A%97%E5%8E%9F/">肿瘤新生抗原</a></li><li class="post-list-item"><a class="post-list-link" href="/2019/12/30/Barnery--JS%E4%B8%AD%E7%9A%84%E5%80%BC%E7%B1%BB%E5%9E%8B/">JS中的值类型</a></li><li class="post-list-item"><a class="post-list-link" href="/2019/12/25/Billy--Jupyter%20notebook%20server/">Jupyter notebook server 搭建</a></li><li class="post-list-item"><a class="post-list-link" href="/2019/10/26/Billy--%E8%A1%A8%E8%A7%82%E7%BB%84%E6%9F%93%E8%89%B2%E4%BD%93%E5%BC%80%E6%94%BE%E5%8C%BA%E5%9F%9F%E5%88%86%E6%9E%90%E2%80%94ATAC-seq/">表观组染色体开放区域分析:ATAC-seq</a></li><li class="post-list-item"><a class="post-list-link" href="/2019/10/05/Billy--%E8%BD%AC%E5%BD%95%E7%BB%84%E5%B7%AE%E5%BC%82%E8%A1%A8%E8%BE%BE%E5%88%86%E6%9E%90%E2%80%94%E2%80%94kallisto/">转录组差异表达分析--免比对工具kallisto</a></li><li class="post-list-item"><a class="post-list-link" href="/2019/09/26/Billy--%E8%BD%AC%E5%BD%95%E7%BB%84%E5%B7%AE%E5%BC%82%E8%A1%A8%E8%BE%BE%E5%88%86%E6%9E%90%E6%A6%82%E8%BF%B0/">转录组差异表达分析概述</a></li><li class="post-list-item"><a class="post-list-link" href="/2019/08/26/Billy--%E4%BD%BF%E7%94%A8poretools%E5%A4%84%E7%90%86fast5%E6%A0%BC%E5%BC%8F%E6%95%B0%E6%8D%AE/">使用poretools处理fast5格式数据</a></li><li class="post-list-item"><a class="post-list-link" href="/2019/07/31/Billy--GATK4%E2%80%94germline%20SNP%20calling/">GATK4：germline SNP calling</a></li></ul></div><div class="widget"><div class="widget-title"><i class="fa fa-gui"> 分类</i></div><ul class="category-list"><li class="category-list-item"><a class="category-list-link" href="/categories/Barney/">Barney</a><span class="category-list-count">2</span></li><li class="category-list-item"><a class="category-list-link" href="/categories/Billy/">Billy</a><span class="category-list-count">14</span></li></ul></div><div class="widget"><div class="widget-title"><i class="fa fa-biao"> 标签</i></div><div class="tagcloud"><a href="/tags/%E7%BA%B3%E7%B1%B3%E5%AD%94%E6%B5%8B%E5%BA%8F/" style="font-size: 15px;">纳米孔测序</a> <a href="/tags/%E7%A5%9E%E7%BB%8F%E7%BD%91%E7%BB%9C/" style="font-size: 15px;">神经网络</a> <a href="/tags/%E4%BA%8C%E4%BB%A3%E6%B5%8B%E5%BA%8F/" style="font-size: 15px;">二代测序</a> <a href="/tags/%E7%BC%96%E7%A8%8B%E8%AF%AD%E8%A8%80/" style="font-size: 15px;">编程语言</a> <a href="/tags/%E8%AE%A1%E7%AE%97%E6%9C%BA%E8%BE%85%E5%8A%A9%E8%8D%AF%E7%89%A9%E8%AE%BE%E8%AE%A1/" style="font-size: 15px;">计算机辅助药物设计</a> <a href="/tags/%E8%BD%AF%E4%BB%B6%E6%95%99%E7%A8%8B/" style="font-size: 15px;">软件教程</a> <a href="/tags/Python/" style="font-size: 15px;">Python</a> <a href="/tags/%E8%A1%A8%E8%A7%82%E7%BB%84/" style="font-size: 15px;">表观组</a> <a href="/tags/%E8%BD%AC%E5%BD%95%E7%BB%84/" style="font-size: 15px;">转录组</a> <a href="/tags/JavaScript/" style="font-size: 15px;">JavaScript</a> <a href="/tags/%E8%82%BF%E7%98%A4/" style="font-size: 15px;">肿瘤</a> <a href="/tags/%E5%8F%98%E5%BC%82%E4%BD%8D%E7%82%B9/" style="font-size: 15px;">变异位点</a></div></div><div class="widget"><div class="widget-title"><i class="fa fa-archive"> 归档</i></div><ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/">2020</a><span class="archive-list-count">3</span></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2019/">2019</a><span class="archive-list-count">13</span></li></ul></div><div class="widget"><div class="widget-title"><i class="fa fa-you"> 友情链接</i></div><ul></ul><a href="https://asia.ensembl.org/index.html" title="ensembl" target="_blank">ensembl</a><ul></ul><a href="https://sci-hub.tw/" title="sci-hub" target="_blank">sci-hub</a></div></div></div></div><a id="totop" href="#top"></a><div id="footer"><div class="footer-info"><p><a href="/baidusitemap.xml">网站地图</a> |  <a href="/atom.xml">订阅本站</a> |  <a href="/about/">联系博主</a></p><p>本站总访问量：<i id="busuanzi_container_site_pv"><i id="busuanzi_value_site_pv"></i></i>次，本站总访客数:<i id="busuanzi_container_site_uv"><i id="busuanzi_value_site_uv"></i></i>人</p><p><span> Copyright &copy;<a href="/." rel="nofollow">Billy &amp; Barney.</a></span><span> Theme by<a rel="nofollow" target="_blank" href="https://github.com/chaooo/hexo-theme-BlueLake"> BlueLake.</a></span><span> Count by<a href="http://busuanzi.ibruce.info/" target="_blank" rel="noopener"> busuanzi.</a></span><span> Powered by<a rel="nofollow" target="_blank" href="https://hexo.io"> Hexo.</a></span></p></div></div></div><script src="https://busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js" async></script><script type="text/javascript" src="/js/search.json.js?v=2.0.4"></script><script type="text/javascript" src="/js/toctotop.js?v=2.0.4" async></script><link rel="stylesheet" type="text/css" href="/share/css/share.css"><script type="text/javascript" src="/share/js/social-share.js" charset="utf-8"></script><script type="text/javascript" src="/share/js/qrcode.js" charset="utf-8"></script></body></html>